### DISCUSSION TITLE
Tailscale basic guide

### DISCUSSION THREAD

**Discussion by akivacp** [gtxaspec/wz_mini_hacks#409] [Q&A] [answered]
Hi, I am a Tailscale user, with basic linux skills. I'm not sure how to set up Tailscale with the camera. I've tried reviewing issue [30](https://github.com/gtxaspec/wz_mini_hacks/issues/30), but I still can't figure it out.
Is there anyone to assist who can help me get started on setting up Tailscale?

**Comment by akivacp** [comment, marked as answer]
**EDIT I updated the instructions so it'll be easy for the next person.** Thank you, @talltechdude for your guidance.

**STEP ONE**
I first generated a Tailscale auth key from here: https://tailscale.com/kb/1085/auth-keys/#generating-a-key

**STEP TWO**
Then I downloaded **Tailscale mipsle** from https://pkgs.tailscale.com/stable/#static. 
I extracted it to `/wz_mini/bin`.

**STEP THREE**
I edited `/wz_mini/etc/cron/root` file and entered this command:
``` 
@reboot tailscaled -statedir /opt/wz_mini/etc/tailscale >> /dev/null 2>&1 &
```

**STEP FOUR**
 In SSH, I ran the command 
```
tailscale up --authkey tskey-abcdef1432341818
```
replacing the auth key with your actual auth key.

Then I rebooted it, tested it and verified.

**Reply by talltechdude** [reply]
@akivacp, your issue with Tailscale giving you a new IP each time is caused by the `statedir` being stored in a tmpfs temporary filesystem location which does not persist across reboots. I've successfully got it working and automatically reconnecting after reboots by changing the crontab line to:
```
@reboot tailscaled -statedir /opt/wz_mini/etc/tailscale >> /dev/null 2>&1 &
```
Then you only need to run the `tailscale up --ssh --authkey tskey-abcdef1432341818` command with your auth-key once via SSH to configure Tailscale (and remove from the crontab file). The `statedir` is then stored on the MMC and re-read in each reboot

PS: also to confirm, the binaries (`tailscale` and `tailscaled`) only need to be in `/opt/wz_mini/bin` dir and not the `tmp` one

**Reply by akivacp** [reply]
Thank you for this answer! This worked!

**Reply by AdnerVL** [reply]
Important to change 
`ENABLE_CRONTAB="true"`
in wz_mini.conf

**Comment by akivacp** [comment]
The only issue I'm experiencing with this is that it creates a new node on the Tailscale network on every connection to the internet instead of connecting to the current node. Would anyone know how to resolve this?

**Comment by akivacp** [comment]
While I do not have a solution to the issue in the previous message, I do have this option:
https://tailscale.com/kb/1111/ephemeral-nodes/

_Ephemeral, for authenticating [ephemeral nodes](https://tailscale.com/kb/1111/ephemeral-nodes) as part of short-lived workloads. Since node keys are not persisted when a workload restarts, these will reconnect as a different node. Nodes which are no longer active will be automatically removed. For example, containers or Lambda functions should use an ephemeral key to connect._

By creating this ephemeral node, the device is deleted after reboot, and I don't have to keep returning to the website to delete. Not the solution I am looking for, but the best with my limited knowledge.

**Comment by akivacp** [comment]
_Update_: because the Tailscale IP keeps changing every time the camera reconnects, I'd like to get notified of the new IP address. Thankfully, there are scripts that post Telegram messages on boot, as one example. 
Here's one user on how to set it up: https://bogomolov.tech/Telegram-notification-on-SSH-login/

Note, the last instruction is to move the script to /etc/profile.d/ folder.
Since I want to get notified on reboot rather than SSH logins, I have my own Telegram folder for my scripts (`/wz_mini/telegramNotif/`) and then I went back to  `/wz_mini/etc/cron/root`, and pasted a command to call the script there.
Here's my example: `@reboot sleep 40 && sh /opt/wz_mini/telegramNotif/login-notify.sh `

As you see from my first post, I gave the camera 30 seconds until after the service starts to then connect to the Tailscale network.
Then, this command, I made it wait ten more seconds so there's enough time to be assigned an IP, and then send the message to Telegram. 

Here are sample files to get started. Note: you will need to remove the TXT extension.
[telegram-send.sh.txt](https://github.com/gtxaspec/wz_mini_hacks/files/10141536/telegram-send.sh.txt)
[login-notify.sh.txt](https://github.com/gtxaspec/wz_mini_hacks/files/10141537/login-notify.sh.txt)



 Actually getting notified on SSH logins are actually a good idea too. However, I can't find that folder.

**Comment by numericOverflow** [comment]
Great work! Tagging along so I can find this guide later
